### PR TITLE
Newest changes

### PR DESCRIPTION
**Pull request by Dimokat0**

### CODE REVIEW

File: src/auth/auth.controller.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  @Post('register')
+  registerUser(
+    @Body('username') username: string,
+    @Body('password') password: string,
+  ) {
+    return this.authService.registerUser(username, password);
+  }
+
+  @Post('login')
+  loginUser(
```
Comment (patatarsk) [inline src/auth/auth.controller.ts]: Better to use dto's for body or parameters.

Like: 

export class LoginUserDto {
  username: string;
  password: string;
}

https://docs.nestjs.com/controllers#request-payloads

Also read about validation.

https://docs.nestjs.com/techniques/validation

Comment (patatarsk) [reply src/auth/auth.controller.ts]: registerUser(
    @Body() payload: LoginUserDto
  )

File: src/auth/auth.repository.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+      user.refresh_token = refreshToken;
+      user.save();
+      return {
+        success: true,
+        accessToken: accessToken,
+        refreshToken: refreshToken,
+        userId: user.id,
+      };
+    }
+    return { success: false, message: 'Wrong password!' };
```
Comment (patatarsk) [inline src/auth/auth.repository.ts]: It's better to throw error => throw new Error() or new BadRequestError().

About exceptions you can read here https://docs.nestjs.com/exception-filters#built-in-http-exceptions

Comment (patatarsk) [reply src/auth/auth.repository.ts]: Instead of return { success: false, message: 'Wrong password!' };

File: src/auth/auth.controller.ts (hunk 1 above)
Comment (Dimokat0) [reply src/auth/auth.controller.ts]: Done it, placed in auth.repository and whenever use it import it from there

File: src/auth/auth.repository.ts (hunk 2 above)
Comment (Dimokat0) [reply src/auth/auth.repository.ts]: Done it also, used BadRequest, UnathorizedException and other in some places

File: src/app.controller.ts [hunk 3]
```diff
@@ -27,6 +27,13 @@ export class AppController {
     res.sendFile(join(__dirname, '..', 'public', 'manage_users.html'));
   }
 
+  @UseGuards(RolesGuard)
+  @SetMetadata('roles', [2, 3])
```
Comment (patatarsk) [inline src/app.controller.ts:31]: Better to use enum instead of magic numbers: export const enum {
  ADMIN = '1',
  USER = '2',
} for example

**Pull request merged**